repositories.md

```diff
@@ -5,10 +5,69 @@ css_id: repositories
+
+* A representation of the public key(s) for the Time Server, similar to the representation of public keys in Root metadata.
+
+Listing the public key of the Time Server in Director targets metadata is necessary to allow partial-verification secondaries to perform time server key rotation.
+
+#### Changes to a Primary
+
+If the Time Server is implemented, the primary SHALL use the following procedure to verify the time. This procedure occurs after the vehicle version manifest is sent and will fulfill the "Download and check current time" step of the Uptane Standard.
+
+1. Gather the tokens from each secondary ECU's version report.
+2. Send the list of tokens to the Time Server to fetch the current time. The time server responds as described in [Time Server](#time_server), providing a cryptographic attestation of the last known time.
+3. If the Time Server's response meets the criteria below, update the primary ECU's clock and retain the Time Server's response for distribution to secondary ECUs, otherwise discard it and proceed without an updated time.  The criteria for checking the Time Server's response are:
+  - The signature over the Time Server's response is valid.
+  - The tokens provided to the Time Server have been included in the response.
+  - The time in the Time Server's response is later than the last time verified in this manner.
+
+#### ECU Version Report
+
+The payload of the ECU version report should contain the latest time downloaded from the Time Server. In addition, the report should include a token (which MUST be used exactly once) for the Time Server to sign and send back.
+
+#### Changes to all ECUs
+
+At build time, ECUs will be provisioned with an attestation of the current time downloaded from the Time Server.
+
+As the first step to verifying metadata, described as "Load and verify the current time or the most recent securely attested time" in the Standard, the ECU SHOULD load and verify the  most recent time from the Time Server using the following procedure:
+
+1. Verify that the signatures on the downloaded time are valid.
+2. Verify that the list of tokens in the downloaded time includes the token that the ECU sent in its previous version report.
+3. Verify that the time downloaded is greater than the previous time.
+
+If all three steps complete without error, the ECU SHALL overwrite its current attested time with the time it has just downloaded, and generate a new token for the next request to the Time Server.
+
+If any check fails, the ECU SHALL NOT overwrite its current attested time, and SHALL jump to the last step ([Create and Send Version Report](https://uptane.github.io/uptane-standard/uptane-standard.html#create_version_report)), and report the error. The ECU MUST reuse its previous token for the next request to the Time Server.
+
+#### Changes to checking Root metadata
```

Comment:
Sorry, @mnm678, just one more thing: you also need to say the same thing for PV secondaries when checking the Targets metadata from the Director repository, no?